x-pack/plugins/cloud_security_posture/server/routes/csp_rule_template/update_csp_rules_states.ts

```diff
@@ -0,0 +1,170 @@
+  return soClient.create<CspSettings>(
+    INTERNAL_CSP_SETTINGS_SAVED_OBJECT_TYPE,
+    {
+      rules_states: {},
+    },
+    { id: INTERNAL_CSP_SETTINGS_SAVED_OBJECT_ID }
+  );
+};
+
+export const createCspSettingObjectSafe = async (
+  soClient: SavedObjectsClientContract,
+  logger: Logger
+) => {
+  const cspSettings = await getCspSettings(soClient, logger);
+
+  if (!cspSettings) return (await createCspSettingObject(soClient)).attributes;
+};
+
+export const getCspSettings = async (
+  soClient: SavedObjectsClientContract,
+  logger: Logger
+): Promise<CspSettings | undefined> => {
+  try {
+    const cspSettings = await soClient.get<CspSettings>(
+      INTERNAL_CSP_SETTINGS_SAVED_OBJECT_TYPE,
+      INTERNAL_CSP_SETTINGS_SAVED_OBJECT_ID
+    );
+    return cspSettings.attributes;
+  } catch (err) {
+    const error = transformError(err);
+    if (error.statusCode === 404) {
+      return undefined;
+    } else {
+      logger.error(`An error occurred while trying to fetch csp settings: ${error}`);
+    }
+  }
+};
+
+export const getCspSettingsSafe = async (
+  soClient: SavedObjectsClientContract,
+  logger: Logger
+): Promise<CspSettings> => {
+  const cspSettings = await getCspSettings(soClient, logger);
+
+  if (!cspSettings) return (await createCspSettingObject(soClient)).attributes;
+
+  return cspSettings;
+};
+
+const buildRuleKey = (benchmarkId: string, benchmarkVersion: string, ruleNumber: string) => {
+  return `${benchmarkId};${benchmarkVersion};${ruleNumber}`;
+};
+
+export const defineUpdateCspRuleStateRoute = (router: CspRouter) =>
+  router.versioned
+    .post({
+      access: 'internal',
+      path: UPDATE_CSP_RULE_STATE_ROUTE_PATH,
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            body: cspRuleBulkActionRequest,
+          },
+        },
+      },
+      async (context, request, response) => {
+        if (!(await context.fleet).authz.fleet.all) {
+          return response.forbidden();
+        }
+        const cspContext = await context.csp;
+
+        try {
+          const requestBody: CspRuleBulkActionRequest = request.body;
+
+          const cspSettings = await getCspSettings(cspContext.soClient, cspContext.logger);
+
+          if (!cspSettings) {
+            throw cspContext.logger.error(`Failed to read csp settings`);
+          }
+
+          const currentRulesStates = cspSettings.rules_states;
+
+          const ruleKeys = requestBody.rules.map((rule) =>
+            buildRuleKey(rule.benchmark_id, rule.benchmark_version, rule.rule_number)
+          );
+
+          const newRulesStates = setRulesStates(
+            currentRulesStates,
+            ruleKeys,
+            muteStatesMap[requestBody.action]
+          );
+
+          const newCspSettings = await updateRulesStates(cspContext.soClient, newRulesStates);
+
+          return response.ok({
+            body: {
+              new_csp_settings: newCspSettings,
```

Comment:
The API here returns SO ⚠️ 
Please fix the response not to directly work with SO